core/src/main/java/com/amplifyframework/core/model/query/QueryOptions.java

```diff
@@ -22,28 +22,46 @@
 import com.amplifyframework.core.model.query.predicate.QueryPredicate;
 import com.amplifyframework.core.model.query.predicate.QueryPredicates;
 
+import java.util.Arrays;
+import java.util.List;
+
 /**
  * A data structure that provides a query construction mechanism that consolidates all query-related
  * options (e.g. predicates, pagination, etc) and allows consumers to build queries in a fluent way.
  */
 public final class QueryOptions {
     private QueryPredicate queryPredicate;
     private QueryPaginationInput paginationInput;
+    private List<QuerySortBy> sortBy;
 
     /**
      * This class should be created using the factory methods such as {@link Where#matchesAll()}
      * and {@link Where#matches(QueryPredicate)}.
      */
     QueryOptions(
             @Nullable QueryPredicate queryPredicate,
-            @Nullable QueryPaginationInput paginationInput
+            @Nullable QueryPaginationInput paginationInput,
+            @Nullable List<QuerySortBy> sortBy
     ) {
         this.queryPredicate = queryPredicate == null ? QueryPredicates.all() : queryPredicate;
         this.paginationInput = paginationInput;
+        this.sortBy = sortBy;
     }
 
     QueryOptions() {
-        this(null, null);
+        this(null, null, null);
+    }
+
+    /**
+     * Update the current options with a given {@code queryPredicate}.
+     *
+     * @param queryPredicate predicate.
+     * @return current options with an updated {@code queryPredicate}.
+     */
+    @NonNull
+    public QueryOptions matches(@NonNull final QueryPredicate queryPredicate) {
```

Comment:
Mostly same discussion on this one as `sorted(...)` I think, except without the list building crap. TLDR, See if you can return a new copy of `QueryOptions` instead of mutating this instance. This will keep the options object immutable.